mindsdb/integrations/handlers/weaviate_handler/weaviate_handler.py

```diff
@@ -0,0 +1,617 @@
+from collections import OrderedDict
+from datetime import datetime
+from typing import List, Optional
+
+import weaviate
+import pandas as pd
+
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+from mindsdb.integrations.libs.response import RESPONSE_TYPE
+from mindsdb.integrations.libs.response import HandlerResponse
+from mindsdb.integrations.libs.response import HandlerResponse as Response
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+from mindsdb.integrations.libs.vectordatabase_handler import (
+    FilterCondition,
+    FilterOperator,
+    TableField,
+    VectorStoreHandler,
+)
+from mindsdb.utilities import log
+from weaviate.util import generate_uuid5
+
+
+class WeaviateDBHandler(VectorStoreHandler):
+    """This handler handles connection and execution of the Weaviate statements."""
+
+    name = "weaviate"
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name)
+
+        self._connection_data = kwargs.get("connection_data")
+
+        self._client_config = {
+            "weaviate_url": self._connection_data.get("weaviate_url"),
+            "weaviate_api_key": self._connection_data.get("weaviate_api_key"),
+        }
+
+        if (
+            self._client_config["weaviate_url"] is None
+            or self._client_config["weaviate_api_key"] is None
+        ):
+            raise Exception(
+                "Both url + api key client secret are required for weaviate connection!"
+            )
+
+        self._client = None
+        self.is_connected = False
+        self.connect()
+
+    def _get_client(self) -> weaviate.Client:
+        client_config = self._client_config
+        if client_config is None:
+            raise Exception("Client config is not set!")
```

Comment:
client_config is dict.
if client_config

Reply:
tried to resolve in b67cb1454fe8803bc304df7430cab6bcd5d670e7